### DISCUSSION TITLE
`.load_history()` does not work on async session

### DISCUSSION THREAD

**Discussion by ehdgua01** [sqlalchemy/sqlalchemy#7339] [Usage Questions] [answered]
### Describe the bug

I tried to get a history of a model instance after flushing,
So I used `.load_history()` on async session like below

```python
result = await session.scalars(select(MyModel).where(ModelModel.id == object_id))
obj = result.one()
obj.name = "changed"
await session.flush()
state = inspect(user_group)

for attr in state.attrs:
    history = attr.load_history()  # raise https://docs.sqlalchemy.org/en/14/errors.html#missinggreenlet
```

however, `.load_history()` raised MissingGreenlet exception!
so, I tried to execute using await, but I got this error
```python
history = await attr.load_history()
TypeError: object History can't be used in 'await' expression
```

### To Reproduce

```python
result = await session.scalars(select(MyModel).where(MyModel.id == object_id))
obj = result.one()
obj.name = "changed"
await session.flush()
state = inspect(obj)
for attr in state.attrs:
    his = await attr.load_history()
```

### Error

```
Traceback (most recent call last):
  File "/Users/A202006042/Desktop/workspace/toy-project/keke.py", line 41, in <module>
    loop.run_until_complete(main())
  File "/Users/A202006042/.pyenv/versions/3.9.4/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
  File "/Users/A202006042/Desktop/workspace/toy-project/keke.py", line 34, in main
    his = await attr.load_history()
TypeError: object History can't be used in 'await' expression
``` 

### Versions

- OS: macos 11.2.1
- Python: 3.9.4
- SQLAlchemy: 1.4.26
- Database: mysql 5.7
- DBAPI (eg: psycopg, cx_oracle, mysqlclient): asyncmy

### Additional context

I currently use `event.listens_for` to solve my problem so not need a solution for my problem. I just contribute for sqlalchemy-orm

**Comment by CaselIT** [comment, marked as answer]
Hi,

Since load_history need to access the db, and there is not yet a async api available the usual solution is to use `run_sync`.

So you can do
```py
def handle_history(sync_session, state):
  for attr in state.attrs:
    his = attr.load_history()
   # do other stuff with sync_session calling it in a sync manner

result = await session.scalars(select(MyModel).where(ModelModel.id == object_id))
obj = result.one()
obj.name = "changed"
await session.flush()
state = inspect(user_group)
await session.run_sync(handle_history, state)
```

more documentation here: https://docs.sqlalchemy.org/en/14/orm/extensions/asyncio.html#running-synchronous-methods-and-functions-under-asyncio

**Comment by ehdgua01** [comment]
@CaselIT
Wow, I didn't know I could use run_sync.
That was the answer.
Thank you.

**Comment by CaselIT** [comment]
let's migrate to a discussion, so maybe it can be of more help to others
